### PR TITLE
Fixed displaying required questions with audio attachments

### LINKED ISSUE #6327: The asterisk is not displayed in required questions if the question contains audio
#### Problem description
The asterisk is not displayed in required questions if the question contains audio. The problem was originally reported on the forum: https://forum.getodk.org/t/required-question-w-audio-not-displaying-red-asterisk/48630
I've also played with other media types (video, image) but it looks like the problem is somehow only related to audio.

#### Steps to reproduce the problem
Open the attached form:
[questionsWithMedia.xlsx](https://github.com/user-attachments/files/16551988/questionsWithMedia.xlsx)

#### Expected behavior
The asterisk should be displayed in all required questions.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #6327 

#### Why is this the best possible solution? Were any other approaches considered?
There was some redundant code updating question labels based on the state of audio attachments (stopped/playing). This code was unnecessary because we only change the color of the labels, not the text. It seems this was needed a long time ago but was never removed. This update didn't account for the state of the questions (such as whether they were required), which was the cause of the issue.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Apart from testing if the issue is fixed. this change requires testing how question labels behave based on the state of their audio attachments. The color of the question label should update accordingly (blue when the audio is playing).

#### Do we need any specific form for testing your changes? If so, please attach one.
Any form with required questions that have audio attachments like the one from the issue.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] added any new strings with date formatting to `DateFormatsTest`
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/RequiredQuestionTest.kt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+@RunWith(AndroidJUnit4::class)
+class RequiredQuestionTest {
+    var rule: CollectTestRule = CollectTestRule()
+
+    @get:Rule
+    var ruleChain: RuleChain = chain()
+        .around(rule)
+
+    @Test
+    fun `requiredQuestionIsMarkedWithAnAsterisk`() {
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/RequiredQuestionTest.kt]: This doesn't need backticks

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/RequiredQuestionTest.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        rule.startAtMainMenu()
+            .copyForm("required_question_with_custom_error_message.xml")
+            .startBlankForm("required_question_with_custom_error_message")
+            .assertText("* Required question")
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/RequiredQuestionTest.kt]: We've already got a helper for this assertion: you can use `assertQuestion("Required question", isRequired = true)`. You can do the same with `swipeToNextQuestion` as well.

**Review by seadowg — CHANGES_REQUESTED** [review]

**Review by seadowg — APPROVED** [review]

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on a device Android 10

**Verified Cases:**
- issue #6327 doesn't occur
- required questions with different media
- forms with required questions and styling examples
- regression checks in forms with required

**Comment by WKobus** [comment]
**Tested with Success**

Verified on device with Android 14

**Pull request merged**